apps/aecore/src/aec_chain.erl

```diff
@@ -593,3 +594,60 @@ get_key_header_by_height(Height) when is_integer(Height), Height >= 0 ->
         error -> {error, chain_too_short};
         {ok, Hash} -> get_header(Hash)
     end.
+
+%%%===================================================================
+%%% Statistics
+%%%===================================================================
+
+-spec sum_tokens_at_height(aec_blocks:height()) ->
+                                  {error, 'chain_too_short'}
+                                | {ok, #{ 'accounts'         => non_neg_integer()
+                                        , 'contracts'        => non_neg_integer()
+                                        , 'contract_oracles' => non_neg_integer()
+                                        , 'locked'           => non_neg_integer()
+                                        , 'oracles'          => non_neg_integer()
+                                        , 'oracle_queries'   => non_neg_integer()
+                                        , 'pending_rewards'  => non_neg_integer()
+                                        , 'total'            => non_neg_integer()
+                                        }}.
+
+
+sum_tokens_at_height(Height) ->
+    %% Wrap in transaction for speed.
+    %% TODO: This could be done dirty
+    aec_db:ensure_transaction(fun() -> int_sum_tokens_at_height(Height)end).
+
+int_sum_tokens_at_height(Height) ->
+    case aec_chain_state:get_key_block_hash_at_height(Height) of
+        error -> {error, chain_too_short};
+        {ok, Hash} ->
+            {ok, Trees} = get_block_state(Hash),
+            Sum = aec_trees:sum_total_coin(Trees),
+            Sum1 = Sum#{pending_rewards => sum_pending_rewards(Height, Hash)},
+            Total = maps:fold(fun(_, X, Acc) -> Acc + X end, 0, Sum1),
+            {ok, Sum1#{ total => Total
+                      , height => Height
```

Comment:
Is this height in return value intended?

Reply:
Good catch! It was intended, but not documented. I will either remove this or document it.

Reply:
If intended the typespec should be updated...

Reply:
I have now removed it